src/ResourceManager/Insights/Commands.Insights/Diagnostics/RemoveAzureRmDiagnosticSettingCommand.cs

```diff
@@ -0,0 +1,69 @@
+﻿// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using System.Management.Automation;
+using System.Net;
+
+namespace Microsoft.Azure.Commands.Insights.Diagnostics
+{
+    /// <summary>
+    /// Get the list of events for at a subscription level.
+    /// </summary>
+    [Cmdlet(VerbsCommon.Remove, "AzureRmDiagnosticSetting", SupportsShouldProcess = true), OutputType(typeof(AzureOperationResponse))]
```

Comment:
@pkamath13 @gucalder is deleting a setting a long-running operation? If so, you should add the [`-AsJob`](https://github.com/Azure/azure-powershell/blob/preview/documentation/development-docs/azure-powershell-developer-guide.md#asjob-parameter) parameter to allow users to run this cmdlet in the background as a PowerShell job

Reply:
It is not a long running operation.